src/dos/drive_local.cpp

```diff
@@ -473,12 +473,12 @@ bool localDrive::FileStat(const char* name, FileStat_Block * const stat_block) {
 	struct stat temp_stat;
 	if (stat(newname,&temp_stat)!=0) return false;
 	/* Convert the stat to a FileStat */
-	struct tm *time;
-	if ((time=localtime(&temp_stat.st_mtime))!=0) {
-		stat_block->time=DOS_PackTime((Bit16u)time->tm_hour,(Bit16u)time->tm_min,(Bit16u)time->tm_sec);
-		stat_block->date=DOS_PackDate((Bit16u)(time->tm_year+1900),(Bit16u)(time->tm_mon+1),(Bit16u)time->tm_mday);
+	struct tm time;
+	if (safe_localtime(&temp_stat.st_mtime, &time)) {
+		stat_block->time = DOS_PackTime(time);
+		stat_block->date = DOS_PackDate(time);
 	} else {
-
+		LOG_MSG("FS: error while converting date in: %s", name);
 	}
```

Comment:
Was thinking we should also provide the problematic `st_mtime` value that caused `safe_localtime(..)` to fail.. but then realized doing so involve yet another unsafe call to `strftime`.. be-getting the need for corresponding `safe_strftime*()` function!  Argh.

Reply:
C++20 provides library for "time of day" and "calendar", but… just by looking at the description of these new interfaces I have no idea how to use them and would prefer to stick to POSIX/C interface instead.

Reply:
Do you know if C++20 deprecates the unsafe POSIX calls in favour of thread-safe POSIX `*_r(...)` variants?

Reply:
C++ committee is completely separate from POSIX; I think POSIX and C standardization bodies have something in common, but C++ is separate. BTW, using old API is still OK when user is 100% certain only a single thread will be ever needed.

The only thing that is truly irritating is that Microsoft ended up having `localtime_s` which is incompatible with C11 `localtime_s` (order of parameters is reversed) - see "Notes" in: https://en.cppreference.com/w/c/chrono/localtime